src/main/java/io/cryostat/net/web/WebServer.java

```diff
@@ -152,6 +152,10 @@ public void start() throws FlightRecorderException, SocketException, UnknownHost
                                         ? exception.getPayload()
                                         : exception.getMessage();
 
+                        if (exception.getCause() != null) {
+                            payload += ": " + exception.getCause().getMessage();
```

Comment:
Nice and simple :-)

I think, from the screenshot you've provided and from what I remember of how I implemented the various exception handling paths that lead here, that this is probably already a sufficient fix for the issue.

But, it might be possible to enhance this message further - it's worth experimenting with at least, although in the end we may just come back to the current solution you have here.

Take a look at this utility class: https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/exception/ExceptionUtils.html

We already have this as a dependency and use this class in a couple of other places.

The methods `getMessage` and `getThrowables` in particular could be neat here to provide even more context around these exceptions.